### PR TITLE
Add PR check for updated metrics docs

### PR DESCRIPTION
**Pull request by machadovilaca**
Signed-off-by: João Vilaça <jvilaca@redhat.com>

**What this PR does / why we need it**:

Validate in each PR that the metrics docs file is up to date

**Which issue(s) this PR fixes**: 

Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: .github/workflows/pull-request.yaml [hunk 1]
```diff
@@ -5,6 +5,19 @@ on:
...
+      - name: Install Golang 1.18
+        uses: actions/setup-go@v2
+        with:
+          go-version: '1.18'
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Generate metrics docs
+        run: make generate-doc
+      - name: Validate no changes
+        run: git difftool -y --trust-exit-code --extcmd=./hack/diff-csv.sh
```
Comment (akrejcir) [inline .github/workflows/pull-request.yaml:20]: Why is the default `git diff` not enough?

Comment (machadovilaca) [reply .github/workflows/pull-request.yaml:20]: using extensions, in this case the ./hack/diff-csv.sh, allows us to ignore certain patterns that might have been changed on some compilation steps, such as the creation date on manifests

Comment (akrejcir) [reply .github/workflows/pull-request.yaml:20]: Ok, is it currently useful? I have tried searching for string `createdAt` in the repo, but it is not anywhere. What manifests do you mean?

Comment (machadovilaca) [reply .github/workflows/pull-request.yaml:20]: Ah, you are right, I tested `make bundle` and checked the `ClusterServiceVersion` generated in the `out/` directory, and unlike, for example, HCO, SSP does not create a `createdAt` annotation. Thank you for the warning, removing

**Comment by akrejcir** [comment]
No need to run e2e tests, because this change is not related to them.

/override ci/prow/e2e-functests
/override ci/prow/e2e-single-node-functests
/override ci/prow/e2e-upgrade-functests

**Pull request merged**